### DISCUSSION TITLE
Set Reminder

### DISCUSSION THREAD

**Discussion by AndriusAlimas** [add2cal/add-to-calendar-button#137] [Ideas / Requests]
Is it possible have feature to set reminder , for example i input in my app custom reminder and when i click any of calendars it will have reminder set in place??

**Comment by jekuer** [comment]
It is not planned yet, because this was considered something, which should be up to the user. There usually are some default settings in place at the used calendar, while it might be considered offensive to some users, while it is also a very unstable feature to the "host" (since this gets easily altered along the way).

However, let's use that discussion to collect up-votes on this issue 😃 .

**Reply by AndriusAlimas** [reply]
What this icsFile property do? I can create generate ics file with reminder included, maybe there is chance pass this ?

**Reply by jekuer** [reply]
It is thought to define a reminder. However, it is ignored by most calendars. Plus, as said, in most cases, it is very bad UX. Defining a reminder should be up to the user, not the organizer.

**Reply by AndriusAlimas** [reply]
If i want define reminder icsFile will be property and what value except is it file url ? can i pass blob:href url . How it gonna work? please and thank you for your answer

**Reply by jekuer** [reply]
I am sorry, but I do not understand what you are trying to say.

**Reply by AndriusAlimas** [reply]
![image](https://user-images.githubusercontent.com/24368155/186027437-e6a5612a-5a48-4386-9b2e-4c56443c06f4.png)

**Reply by AndriusAlimas** [reply]
Lets say i got blob File returned from this method 
![image](https://user-images.githubusercontent.com/24368155/186027615-11d16c24-88f4-49cc-9604-d3f12fc5c8e7.png)
Inside this method return statement like this 
![image](https://user-images.githubusercontent.com/24368155/186027783-2543ebe5-c98c-4a9f-aa0e-85fed4d23881.png)

**Reply by AndriusAlimas** [reply]
If i do window.open(file) i got this reminder set, it will be nice if i can use this variable

**Reply by jekuer** [reply]
Still don't get it. So you are creating your own iCal file. That's awesome! But then why would you need a reminder option? You can simply provide your file and you are good to go.

**Reply by AndriusAlimas** [reply]
![image](https://user-images.githubusercontent.com/24368155/186886519-7dff6810-de0f-4438-bc10-683f59fa2d64.png)
How i can add event listener or event click only when i click on ICal i can use then my file with reminder? everything else is good i only need change for ICal. how to trigger and write my function when i click?

**Reply by jekuer** [reply]
You can set your own iCal file via the `icsFile` option like this:
`"icsFile":"PathToYourIcsFile`

No need to add any event listener. If you really wish to change the core code, have a look at line 508ff at the atcb.js .

**Reply by AndriusAlimas** [reply]
When i console log my file path url blob:http://localhost:3000/229edc99-653d-4319-a960-3b502402224c and is not excepting this format in icsFile i got error : add-to-calendar button generation failed: explicit ics file path not valid

**Reply by jekuer** [reply]
Fair point. It requires .ics as a file extension.
You might need to adjust your blob or add some rewrite Middleware. Also important, since the script will use the filename as filename, which would be quite irritating for the user.

**Reply by AndriusAlimas** [reply]
Thank you for support

**Reply by AndriusAlimas** [reply]
Sorry, where i can find this file atcb.js?I using React tool

**Reply by jekuer** [reply]
There, it would be the index.js in the npm/dist/mjs directory.
But again, it should not be necessary to change the code there for your case (if I understood correctly) - you should rather optimize the blob file and the provide this explicitly via the options to the script. At least that would be the stable way to do.

**Reply by AndriusAlimas** [reply]
Yup i was trying a lot to change blob file, but no luck

**Reply by AndriusAlimas** [reply]
I am nearly there, but i can't re-render Add to calendar button for some reason i using json to show 
![image](https://user-images.githubusercontent.com/24368155/187480241-feb3e10d-15a8-4598-93fb-0f1d0f3b6c37.png)
This is working at first when render 
![image](https://user-images.githubusercontent.com/24368155/187480451-cd4b63ca-8ae4-478d-b707-85d74fc4793b.png)
But when i change reminder i got this 
![image](https://user-images.githubusercontent.com/24368155/187480727-331dafd2-2ae5-4665-882d-d44f3cec070f.png)

**Reply by jekuer** [reply]
Put a wrapper div around the placeholder. Like `<div id="wrapper"><div class="atcb">JSON</div></div>`.
Then, on update, you can simply remove the content of the wrapper-element, put in a new `<div class="atcb">JSON</div>` and re-run the initialization.

**Reply by AndriusAlimas** [reply]
WOW that works !!

**Reply by AndriusAlimas** [reply]
Thank you!
